### PR TITLE
Added docs for the change of behavior of isin

### LINKED ISSUE #38781: DOC: Undocumented change in .isin behavior from 1.1.5 to 1.2.0
#### Location of the documentation
https://pandas.pydata.org/docs/whatsnew/v1.2.0.html

#### Documentation problem
There seems to be an undocumented change with how `isin` works.
pandas 1.1.5:
```python
import pandas as pd
pd.Series([0]).isin(['0']) 
# 0    True
# dtype: bool
pd.Series([1]).isin(['1']) 
# 0    True
# dtype: bool
pd.Series([1.1]).isin(['1.1']) 
# 0    True
# dtype: bool
```
But in pandas 1.2.0 (and 1.2.0rc):
```python
import pandas as pd
pd.Series([0]).isin(['0']) 
# 0    False
# dtype: bool
pd.Series([1]).isin(['1']) 
# 0    False
# dtype: bool
pd.Series([1.1]).isin(['1.1']) 
# 0    False
# dtype: bool
```

But the release notes mentions nothing about it.
I have not posted this issue as an bug,  because I think the newer implementation is the better implementation.
However I propose adding the change to the release notes for other people like me where the change broke tests in a pipeline.

#### Suggested fix for documentation

Add a line mentioning that `isin` no longer returns true when comparing floats to a string of the float.

### PR DESCRIPTION
**Pull request by omarafifii**
This is to close issue #38781, I added documentation to explain the change of behavior of the isin function.

- [x] closes #38781 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by omarafifii** [comment]
I added the docs to the isin in the series.py file. Should I also do the same for the one in frame.py?

File: pandas/core/series.py [hunk 1]
```diff
@@ -4630,6 +4630,34 @@ def isin(self, values) -> "Series":
         4     True
         5    False
         Name: animal, dtype: bool
+        
+        New behaviour from pandas v1.2.0 caused some tests to fail because users where unaware of this change.
```
Comment (phofl) [inline pandas/core/series.py:4634]: I don't think that we need an example from before 1.2 in the docstrings. Just a comment that we do no longer match strings containing numbers with integers and an example from the new behavior? 

Also pls use versionchanged

File: pandas/core/series.py [hunk 2]
```diff
@@ -4630,6 +4630,34 @@ def isin(self, values) -> "Series":
...
+        # dtype: bool
+        pd.Series([1.1]).isin(['1.1']) 
+        # 0    True
+        # dtype: bool
+        
+        From v1.2.0
+        
+        >>>import pandas as pd
+        pd.Series([0]).isin(['0']) 
+        # 0    False
```
Comment (phofl) [inline pandas/core/series.py:4653]: What are the hashes for?

File: pandas/core/series.py [hunk 3]
```diff
@@ -4630,6 +4630,34 @@ def isin(self, values) -> "Series":
...
+        New behaviour from pandas v1.2.0 caused some tests to fail because users where unaware of this change.
+        
+        Before v1.2.0:
+        
+        >>>import pandas as pd
+        pd.Series([0]).isin(['0']) 
+        # 0    True
+        # dtype: bool
+        pd.Series([1]).isin(['1']) 
+        # 0    True
```
Comment (jreback) [inline pandas/core/series.py:4643]: yeah just show the new example (you can add a comment that strings and integers are distinct). Keep the style in-line with the other examples.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (omarafifii) [reply pandas/core/series.py:4634]: Yeah, I was not sure if I should add the before part or not. Ok, I will remove it.
What do you mean by use versionchanged?

File: pandas/core/series.py (hunk 3 above)
Comment (omarafifii) [reply pandas/core/series.py:4643]: Ok thanks for the feedback, I will fix it.

File: pandas/core/series.py (hunk 2 above)
Comment (omarafifii) [reply pandas/core/series.py:4653]: It was meant to be a comment, but you are right, it's better if I remove them.

File: pandas/core/series.py (hunk 1 above)
Comment (phofl) [reply pandas/core/series.py:4634]: .. versionchanged:: 0.24.0

Comment (omarafifii) [reply pandas/core/series.py:4634]: Ok, thanks for the help!

**Comment by simonjayhawkins** [comment]
we may also want to add an entry in the release notes for 1.2 about this change and also a note in the 1.2.1 release notes (other section) that the release notes for 1.2 are updated.

File: pandas/core/series.py [hunk 4]
```diff
@@ -4630,6 +4630,17 @@ def isin(self, values) -> "Series":
         4     True
         5    False
         Name: animal, dtype: bool
+        
+        .. versionchanged:: 1.2.0
```
Comment (jreback) [inline pandas/core/series.py:4634]: you don't neeed the version added.
Also reword this to.

`Strings and integers are distinct and are therefore not comparable`

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 4 above)
Comment (omarafifii) [reply pandas/core/series.py]: Done.
I wanted to ask why are some of the checks failing? Am I doing something wrong?

Comment (jreback) [reply pandas/core/series.py]: click thru to see why these are failign: https://github.com/pandas-dev/pandas/pull/39064/checks?check_run_id=1703810526

for example

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @omarafifii and @simonjayhawkins

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x
